### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by winder**
These changes attempt to provide clearer instructions for how to work with this repository.

Some things, like the "run" script, have been replaced by text in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,78 +16,85 @@ Conduit plugin. This includes:
...
-  exporter_template - Example exporter.
-  file_writer       - Exporter for writing data to a file.
-  noop              - noop exporter
-  postgresql        - Exporter for writing data to a postgresql instance.
+## How to use this repo
+
+The most important step is deciding which type of plugin you should use to
+implement your idea. An `exporter` is good for sending data to a an external
+resource, such as a database or application. `processor` plugins are useful
+for applying rules that filter transactions.
```
Comment (winder) [inline README.md]:
```suggestion
Before anything else, you must decide which type of plugin you should use to
implement your idea. An `exporter` is good for sending data to a an external
resource, such as a database or application. `processor` plugins are useful
for applying rules that filter transactions.
```

File: README.md [hunk 2]
```diff
@@ -26,78 +16,85 @@ Conduit plugin. This includes:
...
+
+### Code
+
+Once you've selected what type of plugin you want to implement, you can start
+writing code. Open this project in your preferred Go IDE and navigate to the
+`plugin` package.
+
+The different life cycle phases are implemented as different functions on the
+interface. For more details on this see the [Development](https://github.com/algorand/conduit/blob/master/docs/Development.md) documentation.
```
Comment (winder) [inline README.md]:
```suggestion
Open the project root in your preferred Go IDE and navigate to the `plugin` package.
Here you'll find templates for the different types of plugins. They have no functionality,
but you'll find some `TODO` comments to identify things you should modify. In particular
you should add configuration parameters and implement the init and data pipeline
functions.

The different life cycle phases are implemented as different functions on the
interface. For more details on this see the [Development](https://github.com/algorand/conduit/blob/master/docs/Development.md) documentation.
```

File: README.md [hunk 3]
```diff
@@ -26,78 +16,85 @@ Conduit plugin. This includes:
...
+
+### Build
+
+A main function is provided at `cmd/conduit/main.go`. It's configured to load
+the plugin templates along with the built-in plugins located in the Conduit
+repository.
+
+To build is a standard operation, run it with: `make conduit`
```
Comment (winder) [inline README.md]:
```suggestion
Build with: `make conduit`
```

File: README.md [hunk 4]
```diff
@@ -26,78 +16,85 @@ Conduit plugin. This includes:
...
+
+### Test
+
+Just like the released version of Conduit, a `config.yml` file is required and
+must be configured. An Algorand node is also required as a data source.
+
+For convenience, consider running your node with docker. This makes it easy to
+configure the network you'll connect to and the API tokens. Modify the
+following command to suite your needs. Note that you can leave out the
+`NETWORK` option to start a private network:
```
Comment (winder) [inline README.md]:
```suggestion
For convenience, consider running your node with docker. It provides methods
to configure the network you'll connect to, setup API tokens, and automatically
configure the node for Conduit. Modify the following command to suite your
needs. Note that you can leave out the `NETWORK` option to start a private
network:
```

File: README.md [hunk 5]
```diff
@@ -1,21 +1,11 @@
...
-4. Test the project: `make run`, and follow the printed instructions.
-
-See the [documentation](https://developer.algorand.org/docs/get-details/conduit/Development/) for more.
-
-![Conduit Quickstart](build/demo/demo.gif)
+Plugin templates are implemented in the `plugin` package. There you'll find
+`importer`, `processor` and `exporter` templates. These have no functionality
+and should be modified.
 
 ## Overview
```
Comment (algoanne) [inline README.md:11]: below: "... to get started with **building** a Conduit plugin."

**Review by algoanne — COMMENTED** [review]
I like the intro framing much better now, it makes it clearer who this is for and what it's for. 

One question as I read it: can I use just one of the offerings and not all 4? e.g. I just want to look at a plugin template file, but not use the suggested release process. 

It also feels slightly strange to me that there's a release process suggested here. Why don't we leave that up to the developer? I assume there's a reason specific to how Conduit works. Also maybe it's just my lack of engineering depth and this is actually totally expected... but if not, we could mention _why_ we added a suggested release process?

**Comment by winder** [comment]
> One question as I read it: can I use just one of the offerings and not all 4? e.g. I just want to look at a plugin template file, but not use the suggested release process.

Hopefully this becomes clear just by using the template. The instructions talk about choosing which one to implement and why you might pick one over another.

> It also feels slightly strange to me that there's a release process suggested here. Why don't we leave that up to the developer? I assume there's a reason specific to how Conduit works. Also maybe it's just my lack of engineering depth and this is actually totally expected... but if not, we could mention _why_ we added a suggested release process?

It's true. This could also link to the 3rd party plugin page: https://github.com/algorand/conduit/blob/master/docs/Third_Party_Plugins.md

**Pull request merged**